src/audio/tdfb/tdfb.c

```diff
@@ -546,35 +540,38 @@ static int tdfb_cmd_enum_get(struct sof_ipc_ctrl_data *cdata, struct tdfb_comp_d
 	return 0;
 }
 
-static int tdfb_cmd_get_value(struct comp_dev *dev, struct sof_ipc_ctrl_data *cdata)
+static int tdfb_cmd_get_value(struct processing_module *mod, struct sof_ipc_ctrl_data *cdata)
 {
-	struct tdfb_comp_data *cd = comp_get_drvdata(dev);
+	struct tdfb_comp_data *cd = module_get_private_data(mod);
 
 	switch (cdata->cmd) {
 	case SOF_CTRL_CMD_ENUM:
-		comp_dbg(dev, "tdfb_cmd_get_value(), SOF_CTRL_CMD_ENUM index=%d", cdata->index);
+		comp_dbg(mod->dev, "tdfb_cmd_get_value(), SOF_CTRL_CMD_ENUM index=%d",
+			 cdata->index);
 		return tdfb_cmd_enum_get(cdata, cd);
 	case SOF_CTRL_CMD_SWITCH:
-		comp_dbg(dev, "tdfb_cmd_get_value(), SOF_CTRL_CMD_SWITCH index=%d", cdata->index);
+		comp_dbg(mod->dev, "tdfb_cmd_get_value(), SOF_CTRL_CMD_SWITCH index=%d",
+			 cdata->index);
 		return tdfb_cmd_switch_get(cdata, cd);
 	}
 
-	comp_err(dev, "tdfb_cmd_get_value() error: invalid cdata->cmd");
+	comp_err(mod->dev, "tdfb_cmd_get_value() error: invalid cdata->cmd");
 	return -EINVAL;
 }
 
-static int tdfb_cmd_set_data(struct comp_dev *dev,
-			     struct sof_ipc_ctrl_data *cdata)
+static int tdfb_get_config(struct processing_module *mod,
+			   uint32_t config_id, uint32_t *data_offset_size,
+			   uint8_t *fragment, size_t fragment_size)
 {
-	struct tdfb_comp_data *cd = comp_get_drvdata(dev);
+	struct sof_ipc_ctrl_data *cdata = (struct sof_ipc_ctrl_data *)fragment;
+	struct tdfb_comp_data *cd = module_get_private_data(mod);
 
-	if (cdata->cmd == SOF_CTRL_CMD_BINARY) {
-		comp_dbg(dev, "tdfb_cmd_set_data(), SOF_CTRL_CMD_BINARY");
-		return comp_data_blob_set_cmd(cd->model_handler, cdata);
-	}
+	/* Fragment size is zero for COMP_CMD_GET_VALUE */
+	if (!fragment_size)
```

Comment:
Note to reviewers:

This check is used to be able to distinguish binary control set/get from other set value controls in module adapter client side. There is no clear way to do it. The code in
 https://github.com/thesofproject/sof/blob/abd81a13836f278fade0987875abd62951f10d8f/src/audio/module_adapter/module_adapter.c#L943

sets fragment _size to cdata->num_elems with binary control blobs. It should never be zero in blob part set/get. The pointer fragment points to cdata->data[0].data, so other parts of the binary control is not made available to client.

Then in 

https://github.com/thesofproject/sof/blob/abd81a13836f278fade0987875abd62951f10d8f/src/audio/module_adapter/module_adapter.c#L1006

The fragment_size is always set to zero. Pointer fragment points to cdata, so the entire ALSA control content is in that case available for client.

Hope this is OK since the module API does not recognize with the header enums any other control types than binary control set/get parts.

Reply:
@ranj063 suggested a different way to accomplish this. I will test it and update the PR and module adapter if looks good.

Reply:
This version is done differently. Now fragment (from module adapter) always points to cdata and the control type can be checked from cdata->cmd.